### PR TITLE
fix for "404 error" on attackless requests and extra slashes 

### PR DESCRIPTION
**Pull request by viskey98**
fixes issue #101

### CODE REVIEW

File: snare.py [hunk 1]
```diff
@@ -216,11 +216,17 @@ def create_data(self, request, response_status):
...
+        
         if detection['type'] == 1:
+            query_start = requested_name.find('?')
+            if query_start != -1:
+                requested_name = requested_name[0:query_start]
+                    
             if requested_name == '/':
                 requested_name = self.run_args.index_page
             try:
+                if requested_name[len(requested_name)-1] == '/':
```
Comment (afeena) [inline snare.py]: can be addressed like requested_name[-1]

**Comment by viskey98** [comment]
@afeena the work on this issue is done , including your suggestions . Please test it and tell me any changes if required :)

File: snare.py [hunk 2]
```diff
@@ -216,11 +217,19 @@ def create_data(self, request, response_status):
...
         status_code = 200
         headers = {}
-
+        p = re.compile('/+')
+        requested_name = p.sub('/',requested_name)
+        
         if detection['type'] == 1:
+            query_start = requested_name.find('?')
+            if query_start != -1:
+                requested_name = requested_name[0:query_start]
```
Comment (afeena) [inline snare.py]: in case you start from 0, you don't need to use lower bound
```
>>> a = [1,2,3,4]
>>> a[:-1]
[1, 2, 3]
```

**Comment by viskey98** [comment]
@afeena changes have been done, please look into it .

**Pull request merged**